### DISCUSSION TITLE
Next.js Eslint

### DISCUSSION THREAD

**Discussion by michalsanger** [adeira/universe#2560] [Ideas]
Next.js v11 integrates Eslint:
https://nextjs.org/docs/basic-features/eslint

Should we include their rules?
https://nextjs.org/docs/basic-features/eslint#eslint-plugin

**Comment by tbergquist-godaddy** [comment]
> Should we include their rules?

Yeah, I think we should do that, we are using nextjs for all our apps, so that makes sense.

**Comment by mrtnzlml** [comment]
I've actually started working on this a long time ago when it was in a beta phase. I will be happy to hand this PR over for someone to take over. One issue I encountered is that we have to somehow make this feature optional because not everything is Next.js. I remember that there were some rules/errors which didn't make any sense in a larger scope as we have the config now. I am pretty sure that was the reason why I didn't finish it yet. 🤔 

Do you want to work on it?

**Reply by michalsanger** [reply]
For sure I want to improve my Next.js app :-) On the other hand how can be the optionality implemented?

**Reply by mrtnzlml** [reply]
I've published my work in progress here: https://github.com/adeira/universe/pull/2562 (let me know if you want to take care of it or no, please). It's actually not that bad when it comes to the errors: all of them are easy to fix or suppress.

> On the other hand how can be the optionality implemented?

I am not really sure, BUT, we might leverage [this](https://github.com/adeira/universe/blob/d9f2714e29c4aed0db7a7d54e938c441d6aebe8d/src/eslint-config-adeira/getCommonConfig.js#L57-L66) which is something other plugins are using for their configuration. We should implement something like this anyway because, for example, we require Jest dependency, but not all projects are using Jest (I have such a project where I have to install Jest only because of this config). We might skip this for now though considering there are just a few Next.js specific errors. 🤔

**Reply by michalsanger** [reply]
How about publish multiple ESLint packages (React, Jest, Node...) and let users use them via `extends` in their config? This way we would not need to add and maintain Next.js rules. And inside Universe we could customise project configuration via `overrides` if needed.

**Reply by mrtnzlml** [reply]
@michalsanger Do you really mean multiple NPM packages or rather what we do with "normal" vs. "strict" rules? For example, now you can do this:

```js
extends: [
  '@adeira/eslint-config', // normal, or:
  '@adeira/eslint-config/strict', // strict
],
```

So you'd be able to do this?

```js
extends: [
  '@adeira/eslint-config', // normal
  '@adeira/eslint-config/strict', // strict
  '@adeira/eslint-config/nextjs',
  '@adeira/eslint-config/react',
  '@adeira/eslint-config/jest',
  // ...
],
```

**Reply by michalsanger** [reply]
Yep exactly as you wrote. 1 NPM package with multiple presets. Including current generic ones to keep it simple for those who don’t need any customisations.

**Reply by mrtnzlml** [reply]
Sounds good. I would be happy to review such PR. 😎

**Reply by michalsanger** [reply]
@mrtnzlml something like this #2569

**Reply by mrtnzlml** [reply]
Nice! I will try to check it as soon as I can. Please, be patient. 🐌

**Reply by mrtnzlml** [reply]
I updated the original PR: https://github.com/adeira/universe/pull/2562

We might need to rethink it a little bit. It's definitely configurable from outside of Universe, however, how do I disable Next.js rules for one repo _inside_ Universe? 🤔

**Reply by michalsanger** [reply]
My idea was to do it via `overrides` in `.eslintrc.js`

**Reply by mrtnzlml** [reply]
Will be released in version 6.3.0 (https://github.com/adeira/eslint-config-adeira/blob/39f2192d5596473cb6ff9c65068747e94e3abc03/CHANGELOG.md#unreleased)
